### DISCUSSION TITLE
Visualization brainstorming

### DISCUSSION THREAD

**Discussion by nickreich** [reichlab/variant-nowcast-hub#78] [Ideas]
Here are the start of a few ideas for visualization tools that we could build for the variant nowcast hub.

It'd be nice to have something interactive that could be re-built every week after the latest forecasts (including the ensemble) are in place.

My thinking is that we might want to have different versions of figure(s) that have x-axis as time and y-axis as clade %. Choices for showing the figures include the following variables:
 - clade
 - location
 - model

So for example, 
 - one figure could be one facet per location showing all clades and one model.
 - one figure could be one facet per model showing all clades for a single location
 - one figure could be one facet per clade showing all models for a given location.
 - ...

**Comment by bsweger** [comment]
Since we're [Reich Lab] running the hub and creating the visualizations, it could be a good opportunity to give [Streamlit](https://streamlit.io/) a spin. The summer experiment was promising.

For hosting, Streamlit provides a [community cloud](https://streamlit.io/cloud) option.

That said, as with any option we choose, we'll need to decide where the data is coming from (s3?) and make sure the data access methods can support the level of performance we want.

**Comment by nickreich** [comment]
Some additional comments from @elray1 on a separate Slack thread:

> Looking at these, I find myself wondering about how well uncertainty is captured in the predictions.  Here are a few thoughts about how we could get at that.
> 
> - 1 page per location, one facet per clade, showing the credible intervals for the mean (the thing that we would submit), for both the period with observed data and continuing to the last prediction horizon
> - 1 page per location, one facet per clade, showing the prediction intervals for the observed data obtained by combining with the multinomial observation process given the daily counts for the data observed so far
> - Some kind of PIT score histogram, again for observed data based on the multinomial, I would think we could start with just looking at each marginal distribution.
